library/src/main/java/com/google/android/media/tv/companionlibrary/EpgSyncJobService.java

```diff
@@ -399,27 +399,30 @@ public Void doInBackground(Void... voids) {
                 }
                 List<Program> programs = getProgramsForChannel(channelUri, channelMap.valueAt(i),
                         startMs, endMs);
-                if (DEBUG) {
-                    Log.d(TAG, programs.toString());
-                }
-                for (int index = 0; index < programs.size(); index++) {
-                    if (programs.get(index).getChannelId() == -1) {
-                        // Automatically set the channel id if not set
-                        programs.set(index,
-                                new Program.Builder(programs.get(index))
-                                        .setChannelId(channelMap.valueAt(i).getId())
-                                        .build());
+
+                if (programs != null) {
```

Comment:
In my case, the provider doesn't have all programs for a given channel in the EPG's XMLTV. Surely enough, I could simply pass an empty list instead of returning null on when  `getProgramsForChannel()` is called